src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -48,53 +57,69 @@ public CartDto getCart() {
     }
 
     @Override
-    public void addToCart(Product product) {
-        product.setTotalInCart(product.getTotalInCart() + 1);
-        List<Product> productList = getCart().getProducts();
-        productList.add(product);
-        getCart().setProducts(productList);
+    public void addToCart(Long id) {
+        List<CartProductDto> contentDtoList = getCart().getCartProductDtoList();
+        if (productNotInCart(id)) {
+            contentDtoList.add(CartProductDto.builder()
+                    .productDto(productMapper.convertToProductDto(productService.findById(id)))
+                    .count(1)
+                    .build());
+        } else {
+            CartProductDto currentCartProductDto = contentDtoList.stream()
+                    .filter(cartProductDto -> cartProductDto.getProductDto().getId().equals(id))
+                    .findFirst()
+                    .get();
+            currentCartProductDto.setCount(currentCartProductDto.getCount() + 1);
+        }
+    }
+
+    private boolean productNotInCart(Long id) {
+        return getCart().getCartProductDtoList().stream()
+                .map(CartProductDto::getProductDto)
+                .map(ProductDto::getId)
+                .filter(i -> i.equals(id))
+                .findFirst()
+                .isEmpty();
     }
 
     @Override
     public void addOneToProductInCart(Long productId) {
```

Comment:
this method is redundant. It's the same scenario you've already covered in the addToCart method. Thus, the getCartContentDtoFromCart method is also redundant. Lots of duplicate code...

Reply:
**second method I mentioned is used three times. So, perhaps better to alter and incorporate. Hint: cutting the stream at findFirst() and working with an Optional

Reply:
removed redundant method before
changed to using optional and orElseThrow() (nicer idd)

Reply:
almost there. Here is what I envisioned: 
@Override
    public void addToCart(Long id) {
        Set<CartProductDto> contentDtoList = getCart().getCartProductDtoSet();
        CartProductDto currentCartProductDto = getCartContentDtoFromCart(id)
                .orElse(CartProductDto.builder()
                        .productDto(productMapper.convertToProductDto(productService.findById(id)))
                        .count(0)
                        .build());
        currentCartProductDto.setCount(currentCartProductDto.getCount() + 1);
        contentDtoList.add(currentCartProductDto);
    }

    private Optional<CartProductDto> getCartContentDtoFromCart(Long productId) {
        return getCart().getCartProductDtoSet().stream()
                .filter(cartCountedProductDto -> productId.equals(cartCountedProductDto.getProductDto().getId()))
                .findFirst();
    }
    
    private Supplier<ProductNotFoundException> productNotFoundExceptionSupplier(Long productId){
        return () -> new ProductNotFoundException("Product with id = " + productId + " was not found in your cart");
    }

    @Override
    public void removeOneFromProductInCart(Long productId) {
        CartProductDto cartProductDto = getCartContentDtoFromCart(productId).orElseThrow(productNotFoundExceptionSupplier(productId));
        if (cartProductDto.getCount() == 1) {
            removeProductFromCart(productId);
        } else {
            cartProductDto.setCount(cartProductDto.getCount() - 1);
        }
    }
NOTE: that I removed the addOneToProductInCart() method, since it perfectly safe(r) to use the addToCart method instead. So the only thing now, if you were to implement the refactored code, is to change the method in the UserController. 
If not totally refactoring, then I still think you should at least remove the addOneToProductInCart() method.

Reply:
fixed and got to know the Suplier<ProductNotFoundException> .... never seen that

Reply:
always look for the IntelliJ hints. It tells you what it expects. Also: don't blindly copy-paste please 😉 My quick refactoring left out the Set's objectType. Fixed it and will push.